tests/test_qnode.py

```diff
@@ -1398,6 +1348,488 @@ def circuit(a):
             circuit.jacobian([1.0], method="A")
 
 
+class TestMetricTensor:
+    """Tests for metric tensor subcircuit construction and evaluation"""
+
+    def test_no_generator(self):
+        """Test exception is raised if subcircuit contains an
+        operation with no generator"""
+        dev = qml.device('default.qubit', wires=1)
+
+        def circuit(a):
+            qml.Rot(a, 0, 0, wires=0)
+            return qml.expval(qml.PauliX(0))
+
+        circuit = qml.QNode(circuit, dev)
+
+        with pytest.raises(QuantumFunctionError, match="has no defined generator"):
+            circuit.construct_metric_tensor([1])
+
+    def test_generator_no_expval(self, monkeypatch):
+        """Test exception is raised if subcircuit contains an
+        operation with generator that corresponds to no expectation value"""
+        dev = qml.device('default.qubit', wires=1)
+
+        def circuit(a):
+            qml.RX(a, wires=0)
+            return qml.expval(qml.PauliX(0))
+
+        circuit = qml.QNode(circuit, dev)
+
+        with monkeypatch.context() as m:
+            m.setattr('pennylane.RX.generator', [qml.RX, 1])
+
+            with pytest.raises(QuantumFunctionError, match="no corresponding expectation value"):
+                circuit.construct_metric_tensor([1])
+
+    def test_construct_subcircuit(self):
+        """Test correct subcircuits constructed"""
+        dev = qml.device('default.qubit', wires=2)
+
+        def circuit(a, b, c):
+            qml.RX(a, wires=0)
+            qml.RY(b, wires=0)
+            qml.CNOT(wires=[0, 1])
+            qml.PhaseShift(c, wires=1)
+            return qml.expval(qml.PauliX(0))
+
+        circuit = qml.QNode(circuit, dev)
+
+        circuit.construct_metric_tensor([1, 1, 1])
```

Comment:
You don't want to include parameter values in the `assert`s?

Reply:
How do you mean?

Reply:
Don't the subcircuits also store a scale parameter for the generator? Shouldn't we check whether those are correctly extracted, stored, and used?

Reply:
Added